sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -659,11 +661,36 @@ def __init__(self, source, test_bigquery_client=None, use_legacy_sql=True,
     else:
       self.query = self.source.query
 
+  def _get_source_query_location(self):
+    referenced_locations = self.client.get_referenced_locations(
+        self.executing_project,
+        self.source.query,
+        use_legacy_sql=self.source.use_legacy_sql)
+
+    logger.debug("Referenced locations: %r", referenced_locations)
+
+    locations = set(referenced_locations.values())
+
+    if len(locations) > 1:
```

Comment:
I have to double check, but I believe that Dataflow will pick the first location it finds if there is more than 1 location.

Reply:
@chamikaramj says that BigQuery shouldn't allow queries with tables in more than 1 location so len(locations) should always be 1, but it doesn't hurt to check.

Reply:
For example, the Java SDK takes the location of the first referenced table:
https://github.com/apache/beam/blob/279a05604b83a54e8e5a79e13d8761f94841f326/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQuerySource.java#L117-L120

I believe that is sufficient. Also note that the docs for referencedTables states:
> Queries that reference more than 50 tables will not have a complete list.

Reply:
Fixed in https://github.com/apache/beam/pull/6463/commits/560070ce3d632d77039cfa13259d213fdfdd1c3f